core/src/main/scala/pipeflow/system/PipeFlowSystem.scala

```diff
@@ -0,0 +1,55 @@
+package pipeflow.system
+
+import java.time.LocalDateTime
+
+import akka.actor.ActorSystem
+import com.typesafe.config.{Config, ConfigFactory}
+import org.slf4j.LoggerFactory
+import pipeflow.dsl.nodes.Node
+import pipeflow.system.iso8601.RepeatingInterval
+
+import scala.concurrent.duration.{Duration, FiniteDuration}
+import scala.util.{Failure, Success}
+
+
+object PipeFlowSystem {
+  type NodeBuilder = (LocalDateTime) => Node
+
+  private val logger = LoggerFactory.getLogger(PipeFlowSystem.getClass.getName.split("[.$]").last)
+
+  private class ScheduleRepeatingIntervalException(msg: String) extends Exception(msg)
+
+  def apply(name: String): PipeFlowSystem = new PipeFlowSystem(name, ConfigFactory.load())
+}
+
+class PipeFlowSystem(val name: String, val config: Config) {
+
+  import PipeFlowSystem.{logger, NodeBuilder, ScheduleRepeatingIntervalException}
+
+  val system = ActorSystem(name, config)
+
+  def schedule(iso8601RepeatingInterval: String)(nodeBuilder: NodeBuilder): Unit = {
+    RepeatingInterval(iso8601RepeatingInterval) match {
+      case Right(repeatingInterval) => schedule(repeatingInterval)(nodeBuilder)
+      case Left(msg) => throw new ScheduleRepeatingIntervalException(msg)
```

Comment:
@SleimanJneidi how would you get rid of this exception ? Given that this function returns a Unit, I don't see how Either could help very much here.

Reply:
Unit is an anti pattern, `schedule ` should return an Either I would argue

Reply:
An Either of what ? The PipeFlowSystem is mutable, and the schedule function modifies its state (by creating an actor that will take care of building a graph according to the repeating interval), there is nothing to be returned right now (although I have some ideas for future iterations), that's why it returns a Unit.

Reply:
I see, we can return an Id of the scheduler or the Actor itself, so for now we can keep the exception

Reply:
Fine. I was also thinking in returning either an Id, or a custom Cancellable trait for the schedule, but I'd still like to think a bit more this interface. About returning an Actor, I was thinking to consider it an implementation detail, and not to expose any detail about the actor system through the API.